spec/schedulers/AnimationFrameScheduler-spec.ts

```diff
@@ -141,6 +141,46 @@ describe('Scheduler.animationFrame', () => {
     }
   });
 
+  it('should schedule next frame actions from a delayed one', (done) => {
+    animationFrame.schedule(() => {
+      animationFrame.schedule(() => { done(); });
+    }, 1);
+  });
+
+  it('should schedule 2 actions for a subsequent frame', (done) => {
+    let runFirst = false;
+    animationFrame.schedule(() => {
+      animationFrame.schedule(() => { runFirst = true; });
+      animationFrame.schedule(() => {
+        if (runFirst) {
+          done();
+        } else {
+          done(new Error('First action did not run'));
+        }
+      });
+    });
+  });
+
+  it('should handle delayed action without affecting next frame actions', (done) => {
+    let runDelayed = false;
+    let runFirst = false;
+    animationFrame.schedule(() => {
+      animationFrame.schedule(() => { runFirst = true; });
+      animationFrame.schedule(() => {
+        if (!runDelayed) {
+          done(new Error('Delayed action did not run'));
+        } else if (!runFirst) {
+          done(new Error('First action did not run'));
+        } else {
+          done();
+        }
+      });
+
+      // This action will execute before the next frame because the delay is less than the one of the frame
+      animationFrame.schedule(() => { runDelayed = true; }, 1);
```

Comment:
I'm curious which order in which these actions are expected to execute.

Is `runFirst` expected to be true when `runDelayed` is set to true? e.g.:

```suggestion
      animationFrame.schedule(() => {
        if (!runFirst) {
          done(new Error('First action did not run'));
        } else {
          runDelayed = true;
        }
      }, 1);
```

Reply:
No, it's the opposite. The action queued with a delay smaller than animationFrame runs first.
That's how animationFrameScheduler is implemented, actions with delay other than zero are scheduled purely based on time and not actual animation frames.
I'll push a change to make it more clear.

Reply:
There. I pushed the change. Now each scheduled actions asserts the previous one. Instead of having everything in the last one.